Updated aws-ec2-inventory docs (#2625)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cse-aws-ec-inventory-source.md

The [IAM policy](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-policies-for-amazon-ec2.html) needs the `ec2:DescribeInstances` and `ec2:DescribeImages` permissions.

## States

> A CSE AWS EC2 Inventory Source tracks errors, reports its health, and start-up progress. You’re informed, in real-time, if the Source is having trouble connecting, if there's an error requiring user action, or if it is healthy and collecting by utilizing Health Events.
> 
> A CSE AWS EC2 Inventory Source goes through the following states when created:

Reviewer: ```suggestion
The CSE AWS EC2 Inventory Source tracks errors, reports its health, and start-up progress. You’re informed, in real-time, if the Source is having trouble connecting, if there's an error requiring user action, or if it is healthy and collecting by utilizing Health Events.

The CSE AWS EC2 Inventory Source goes through the following states when created:
```

1. **AWS Access**. The integration is configured for either role based AWS authentication or key based AWS authentication.
    - **Role Based Access**. AWS Role ARN is required for Role based Access. Use the information provided on the source page to configure the role.<br/><img src={useBaseUrl('/img/send-data/role-based.png')} alt="role-based" style={{border: '1px solid black'}} width="400"/>
    - **Key Access**. Enter the IAM user access key ID and secret key you want to use to authenticate collection requests.<br/><img src={useBaseUrl('/img/send-data/key-based.png')} alt="key-based" style={{border: '1px solid black'}} width="400"/>
1. **Regions**. Provide a list of AWS regions to query EC2 instances, such as `us-east-2`.
1. (Optional) The **Polling Interval** is set for 600 minutes by default, you can adjust it based on your needs.

> 1. **Processing Rules for Logs (Optional)**. Configure any desired filters, such as allowlist, denylist, hash, or mask, as described in Create a Processing Rule.

Reviewer 2: ```suggestion
1. **Processing Rules for Logs (Optional)**. Configure any desired filters, such as allowlist, denylist, hash, or mask, as described in [Create a Processing Rule](/docs/send-data/collection/processing-rules/create-processing-rule/).
```

| Parameter | Type | Required | Description | Access |
|:--|:--|:--|:--|:--|

> | config | JSON Object | Yes | Contains the configuration parameters for the Source. | na |

Reviewer: ```suggestion
| config | JSON Object | Yes | Contains the configuration parameters for the Source. | n/a |
```

| `fields` | JSON Object | No |  | JSON map of key-value fields (metadata) to apply to the Collector or Source. Use the boolean field _siemForward to enable forwarding to SIEM. | modifiable |
| `limitToRegions` | String array | Yes | | Provide a list of AWS regions to query EC2 instances, such as `us-east-2`. Use `["all"]` to support all regions. | modifiable |
|`authentication.type` | String | Yes | AWS Role Based Authentication | Select how Sumo Logic should access your AWS account. | modifiable |
|`authentication.roleARN` | String | Yes | |Provide the IAM Role ARN you want to use to authenticate collection requests. | modifiable |
|`authentication.awsId` | String | Yes | | Provide the IAM User access key ID you want to use to authenticate collection requests. | modifiable |
|`authentication.roleARN` | String | Yes | |Provide the Secret Key you want to use to authenticate collection requests. | modifiable |
|`polling_interval` | Integer | No | 600 | The minutes in between checks for new data. | modifiable |

### JSON Example

> ```json title="JSON example for Role Based Authentication"

Reviewer: ```suggestion
```json title="JSON example for Role-Based Authentication"
```
